src/lib.rs

```diff
@@ -321,13 +370,10 @@ impl<T> Entry<T> {
 
     #[inline]
     fn remove(&mut self, nxt: usize) -> Option<T> {
-        if self.in_use() {
-            return match mem::replace(self, Entry::Empty(nxt)) {
-                Entry::Filled(val) => Some(val),
-                Entry::Empty(_) => unreachable!(),
-            }
+        match mem::replace(self, Entry::Empty(nxt)) {
+            Entry::Filled(val) => Some(val),
+            Entry::Empty(_) => None
         }
-        None
     }
```

Comment:
I'm pretty sure removing the check for `in_use` will leak `Entry` blocks.

`mem::replace(Empty(some_index_to_empty_location), Empty(nxt))`

Without saving `some_index_to_empty_location`, that "pointer" will be lost forever.
There really should be a test case that ensures `remove` doesn't leak